### ISSUE TITLE
`remark-lint-strikethrough-marker` rule not working

### ISSUE THREAD

**Issue by jrfnl** [remarkjs/remark-lint#291]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-lint-strikethrough-marker

### Link to runnable example

https://github.com/jrfnl/bug-report-reproduction-scenarios/tree/remark-lint/strike-through-rule

### Steps to reproduce

1. Given that the `remark-cli`, `remark-lint` and `remark-lint-strikethrough-marker` packages are installed ...

2. And there is a `.remarkrc` config file containing the following:
```json
{
  "plugins": [
    ["remark-lint-strikethrough-marker", "~~"]
  ]
}
```
(the option value could also be set to `consistent` or left out, same problem occurs).

3. And given the following markdown sample:
```md
# Test document

This is striken ~~foo~~ through.
This is striken ~foo~ through.
```

4. When running the command `remark ./test.md --frail`...

### Expected behavior

I'd expect an error on line 4 saying `Strikethrough should use `~~` as a marker`

### Actual behavior

Remark lint reports no issues found.

GH Actions workflow output demonstrating this behaviour: https://github.com/jrfnl/bug-report-reproduction-scenarios/runs/6026236264?check_suite_focus=true

---

Related PR introducing the rule: https://github.com/remarkjs/remark-lint/pull/260

### Runtime

Node v16

### Package manager

npm 8

### OS

Linux

**Comment by jrfnl** [comment]
@wooorm Sorry, but how does that translate to a "wontfix" ? I tried the above with `remark-gfm` installed as well (before reporting it) and the error still doesn't show up.

It's perfectly fine to tell me I'm doing something wrong and it's not a bug, but please use words....

**Comment by wooorm** [comment]
The docs say that `remark-gfm` is needed. your repro doesn’t use `remark-gfm`. The fix to your repro is using `remark-gfm`.
If there is a different problem, please add `remark-gfm` to your example.

**Comment by jrfnl** [comment]
@wooorm Thanks for clarifying.

I'd tried with the `remark-gfm` package installed before, but didn't add it to the `.remarkrc` file. I've now tried with that additional change and yes, now the error does show up.

Thanks for pointing me in the right direction.

As a side-note: as experienced dev, but one who doesn't do much JS dev, I found it quite challenging to get Remark-lint running. I read all the relevant docs for NPM, Remark-CLI, Remark-Lint, but was still struggling (a lot).

Something like the above issue (and the solution) is just an example of where the docs presume a lot of contextual knowledge of the user, which the user may not have.

It might help future users to have a more entry-level "how to get started" document or possibly some links to tutorials elsewhere on the web which explain things ?

**Comment by wooorm** [comment]
Thanks for the feedback. unified is indeed complex. It’s also something that will always be complex: it’s several hundreds of github repos and something like 700 npm packages. And it’s all maintained by very few people.

That is to say, it will never be easy.
That is not to say, the docs cannot be improved. Please suggest improvements on how it can be made more clear that `remark-gfm` needs to be used like other plugins are used, and please write such beginners tutorials and we can link to them from here.

**Comment by jrfnl** [comment]
> Thanks for the feedback. unified is indeed complex. It’s also something that will always be complex: it’s several hundreds of github repos and something like 700 npm packages. And it’s all maintained by very few people.

I understand and appreciate that. Am in a similar situation with plenty of my projects.

> That is to say, it will never be easy. That is not to say, the docs cannot be improved.

The circle of open source: I was trying to implement Remark as a part of the QA for the docs of one of the open source projects I'm working on ;-)
The idea being that if I could get it working in that project, I'd have a "template" for how to implement it in more projects.

>  Please suggest improvements on how it can be made more clear that `remark-gfm` needs to be used like other plugins are used

Some of the things I learned about the NPM/JS/Remark eco-system from trying to get things working:
* To be able to use the `remark` command from the command-line, `remark-cli` needs to be installed as `--global` as otherwise the command will not be available.
* Installing `remark-lint` will not install any of the rules which are maintained in the project.
* If any of the presets are installed globally, the individual rules included via the preset are not available to a local project unless you reference the preset in the `.remarkrc` file.
* If the preset(s) is/are installed locally, the individual rules _will_ be available to a local project without needing to reference the preset in the `.remarkrc` file.
* There is no preset available which includes **all** the rules maintained in the `remark-lint` project.
* The combination of all three presets doesn't include all the rules maintained in the `remark-lint` project, so you will need to manually check if a rule needs to be installed separately or if it's included in a preset.
* If a rule says it works with `remark-gfm`, `remark-gfm` not only needs to be installed, it also needs to be added to the `.remarkrc` file.
* When an option expects an object and you provide the object via `.remarkrc`, you need to put quotes around the object and pass the object as a string. I suspect the same applies to arrays ?

Pet-peeve (totally not related to this project, but to NPM):
There doesn't appear to be a `verbose` option which shows which packages were installed - or at least: I haven't been able to find one.
This made debugging the above that much harder.

A lot of these things feel like they will probably be obvious to someone who is used to the JS/NPM infrastructure/dependency management system, but coming from a PHP/Composer background, they were very surprising and non-obvious to me.



> and please write such beginners tutorials and we can link to them from here.

Sorry, I don't blog. That's aside from the fact that I for sure don't have enough contextual knowledge to write a tutorial.

**Comment by wooorm** [comment]
I understand your use case and you’re not the first. Many other people come in with a particular wish such as “I want to do X with markdown” (e.g., lint, add a table of contents, turn it into markdown). unified is much more than that. You can check *and* format *and* transform your content.

I believe most of your insights stem from not knowing npm/the javascript ecosystem/related tools. That’s super understandable, I understand each point. You’ll find similar things when using Babel, webpack, ESLint, etc though. I don’t think it makes sense to duplicate those docs here. I don’t want to teach people how npm, Node.js, JavaScript, and the ecosystem work: there are far better sources for that!
For debugging info, npm has a flag for that (`--loglevel`), and there are certain shorthands: https://docs.npmjs.com/cli/v8/using-npm/config#shorthands-and-other-cli-niceties.

**Comment by jrfnl** [comment]
> For debugging info, npm has a flag for that (`--loglevel`), and there are certain shorthands: https://docs.npmjs.com/cli/v8/using-npm/config#shorthands-and-other-cli-niceties.

Thanks for that link!

Interesting, though - to me - again very counter-intuitive. Based on how I read those docs, it looks like I have to run a `npm config ...` command ahead of a `npm install` to use it. It also talks about writing everything to a log _file_ instead of showing it on `STDOUT` which is what I'd need as in CI access to files like that is generally not easy.

Also, from a docs perspective a sentence like this is unclear as there is no information making it explicit whether the list of options shown is in any particular order. What is "higher" ?
> Any logs of a higher level than the setting are shown. The default is "notice".

I'd found the `foreground-scripts` option mentioned in the `npm install` docs and had used that in one of my attempts, but that never seemed to make a difference.

Either way, thanks for your time and your pointers. I appreciate the work you do here!
